### PR TITLE
Compute v2: Add the user id to the get create and delete keypair

### PR DESCRIPTION
**Pull request by jtopjian**
For #2145 
Supersedes #2146

### CODE REVIEW

**Comment by jtopjian** [comment]
@kayrus Let me know your thoughts on this. Generally, I'm not opposed to adding an `opts` field to existing functions, but since Key Pairs are part of the earliest and foundational resources in Gophercloud, I want to be careful and retain backwards compatibility.

File: openstack/compute/v2/extensions/keypairs/requests.go [hunk 1]
```diff
@@ -81,9 +85,85 @@ func Get(client *gophercloud.ServiceClient, name string) (r GetResult) {
...
+// ToKeyPairGetQuery formats a GetOpts into a query string.
+func (opts GetOpts) ToKeyPairGetQuery() (string, error) {
+	q, err := gophercloud.BuildQueryString(opts)
+	return q.String(), err
+}
+
+// GetWithOpts is similar to Get, but can take a GetOpts struct.
+// The options within GetOpts require the client to have a microversion set.
+// The original Get function is retained for non-microverisoned requests.
+func GetWithOpts(client *gophercloud.ServiceClient, name string, opts GetOptsBuilder) (r GetResult) {
```
Comment (kayrus) [inline openstack/compute/v2/extensions/keypairs/requests.go:110]: @jtopjian `GetWithOpts` func looks hacky. I'd extend the `Get` function, even if it breaks the previous API.

**Comment by kayrus** [comment]
@jtopjian can you also add a `UserID` parameter to the `List` function?

**Comment by kayrus** [comment]
@jtopjian thanks a lot. LGTM.

**Pull request merged**